expo-app-auth-deprecated.md

```diff
@@ -0,0 +1,159 @@
+
+  return null;
+}
+```
+
+### `AppAuth.authAsync`
+
+The `authAsync` method contains a lot of complexity for things like auto exchanging client side secrets (never store your secrets in the client!). This functionality has been divided across a couple of different methods and classes in `expo-auth-session`. It is highly recommended that you use the React Hooks API for performing authentication in `expo-auth-session` via `AuthSession.useAuthRequest()`.
+
+For examples of performing authentication with `expo-auth-session`, check out the [Authentication guide](https://docs.expo.dev/guides/authentication/).
+
+Example of [auto exchanging client secret](https://docs.expo.dev/guides/authentication/#coinbase) on the client side (dangerously) in the Coinbase "Auth Code" demo.
+
+For even more info, check out the [API reference](https://docs.expo.dev/versions/latest/sdk/auth-session/#useauthrequest).
+
+### `AppAuth.revokeAsync` -> `AuthSession.revokeAsync`
+
+> [`AuthSession.revokeAsync()` API](https://docs.expo.dev/versions/latest/sdk/auth-session/#authsessionrevokeasync).
+
+The `revokeAsync` method wraps an request to the `revocationEndpoint` URL. This is a simple system to migrate between.
+
+```ts
+const clientId = "XXX";
+const token = "XXX";
+const serviceConfiguration = {
+  revocationEndpoint: "...",
+};
+
+AppAuth.revokeAsync(
+  {
+    clientId,
+    serviceConfiguration,
+  },
+  {
+    token,
+    isClientIdProvided: !!clientId,
+  }
+);
+
+// v v v
+
+AuthSession.revokeAsync(
+  {
+    clientId,
+    token,
+  },
+  serviceConfiguration
+);
+```
+
+### `AppAuth.refreshAsync` -> `AuthSession.refreshAsync`
+
+> [`AuthSession.refreshAsync()` API](https://docs.expo.dev/versions/latest/sdk/auth-session/#authsessionrefreshasync).
+
+The `refreshAsync` method wraps an request to the `tokenEndpoint` URL. This is a simple system to migrate between. Spec: [Section 6](https://tools.ietf.org/html/rfc6749#section-6).
+
+```ts
+const clientId = "XXX";
+const refreshToken = "XXX";
+const serviceConfiguration = {
+  tokenEndpoint: "...",
+};
+
+AppAuth.refreshAsync(
+  {
+    clientId,
+    serviceConfiguration,
+    isRefresh: true,
+  },
+  refreshToken
+);
+
+// v v v
+
+AuthSession.refreshAsync(
+  {
+    clientId,
+    refreshToken,
+  },
+  serviceConfiguration
+);
+```
+
+### `AppAuth.URLSchemes`
+
+There is currently no replacement for this iOS-only constant which returns the URI schemes from the app's native `Info.plist`. This may be added to `expo-linking` or `expo-application` in the future.
+
+Check out the [redirect URI patterns](https://docs.expo.dev/guides/authentication/#redirect-uri-patterns) guide for more info.
+
+### `AppAuth.OAuthRedirect`
+
+There is currently no replacement for this constant which simply indicates a native URI scheme that maps to the bundle identifier on iOS and the package name on Android.
+
+This can be generally retrieved by using `expo-application`:
```

Comment:
why "generally"? what exceptions are there?

Reply:
Android package name ≠ applicationId but we use the package name for the uri scheme.